mppsolar/outputs/prom.py

```diff
@@ -0,0 +1,91 @@
+import logging
+import re
+
+from .baseoutput import baseoutput
+from ..helpers import get_kwargs, key_wanted
+
+log = logging.getLogger("prom")
+
+
+class prom(baseoutput):
+    def __str__(self):
+        return "outputs Node exporter prometheus format to standard out"
+
+    def __init__(self, *args, **kwargs) -> None:
+        log.debug(f"processor.value __init__ kwargs {kwargs}")
+
+    def printHeader(command, description):
+        pass
+
+    def output(self, *args, **kwargs):
+        log.info("Using output processor: value")
+        log.debug(f"kwargs {kwargs}")
+        tag = get_kwargs(kwargs, "tag")
+        dev = get_kwargs(kwargs, "name")
+        data = get_kwargs(kwargs, "data")
+        name = get_kwargs(kwargs, "name")
+        if data is None:
+            return
+
+        # check if config supplied
+        config = get_kwargs(kwargs, "config")
+        if config is not None:
+            log.debug(f"config: {config}")
+            # get formatting info
+            remove_spaces = config.get("remove_spaces", True)
+            keep_case = config.get("keep_case", False)
+            filter = config.get("filter", None)
+            excl_filter = config.get("excl_filter", None)
+            tag = config.get("tag", None)
+            name = cconfig.get("name", "mpp_solar")
+
+        else:
+            # get formatting info
+            remove_spaces = True
+            keep_case = get_kwargs(kwargs, "keep_case")
+            filter = get_kwargs(kwargs, "filter")
+            excl_filter = get_kwargs(kwargs, "excl_filter")
+            tag = get_kwargs(kwargs, "tag")
+            name = get_kwargs(kwargs, "name")
+
+        if filter is not None:
+            filter = re.compile(filter)
+        if excl_filter is not None:
+            excl_filter = re.compile(excl_filter)
+        if name == "unnamed":
+            name = "mpp_solar"
+        # remove raw response
+        if "raw_response" in data:
+            data.pop("raw_response")
+
+        dev = data.pop("_command", None)
```

Comment:
this should go into the `if "_command" in data block and have a default for the case where the if is false

Reply:
I may need an example, If I move it anywhere other than where it is, the output is always None.
Which is okay with prometheus, though be nice to have expected function. Sample output with what I just posted:
`/usr/lib/python3.11/site-packages/mppsolar/outputs # mpp-solar -c  QPGS2 -P pi30max -o prom2
mpp_solar_parallel_instance_number{device="None",cmd="QPGS2",myStr="valid"} 0
mpp_solar_serial_number{device="None",cmd="QPGS2",myStr="96xxxxxxxxxxxx"} 0
mpp_solar_work_mode{device="None",cmd="QPGS2",myStr="Battery Mode"} 0
mpp_solar_fault_code{device="None",cmd="QPGS2",myStr="No fault"} 0
mpp_solar_grid_voltage{device="None",cmd="QPGS2"} 121.8
mpp_solar_grid_frequency{device="None",cmd="QPGS2"} 59.99
mpp_solar_ac_output_voltage{device="None",cmd="QPGS2"} 119.9`

Reply:
Sample using name:
` # mpp-solar -c  QPGS2 -P pi30max -o prom2 -n LV6548_P1
LV6548_P1_parallel_instance_number{device="None",cmd="QPGS2",myStr="valid"} 0
LV6548_P1_serial_number{device="None",cmd="QPGS2",myStr="96342112100900"} 0
LV6548_P1_work_mode{device="None",cmd="QPGS2",myStr="Battery Mode"} 0
LV6548_P1_fault_code{device="None",cmd="QPGS2",myStr="No fault"} 0
LV6548_P1_grid_voltage{device="None",cmd="QPGS2"} 121.8
LV6548_P1_grid_frequency{device="None",cmd="QPGS2"} 60.01
LV6548_P1_ac_output_voltage{device="None",cmd="QPGS2"} 119.6
LV6548_P1_ac_output_frequency{device="None",cmd="QPGS2"} 60.0
LV6548_P1_ac_output_apparent_power{device="None",cmd="QPGS2"} 1746
LV6548_P1_ac_output_active_power{device="None",cmd="QPGS2"} 1654
LV6548_P1_load_percentage{device="None",cmd="QPGS2"} 25
LV6548_P1_battery_voltage{device="None",cmd="QPGS2"} 51.9
LV6548_P1_battery_charging_current{device="None",cmd="QPGS2"} 0
LV6548_P1_battery_capacity{device="None",cmd="QPGS2"} 56
LV6548_P1_pv1_input_voltage{device="None",cmd="QPGS2"} 98.7
LV6548_P1_total_charging_current{device="None",cmd="QPGS2"} 0
LV6548_P1_total_ac_output_apparent_power{device="None",cmd="QPGS2"} 5674
LV6548_P1_total_output_active_power{device="None",cmd="QPGS2"} 5459
LV6548_P1_total_ac_output_percentage{device="None",cmd="QPGS2"} 41
LV6548_P1_is_scc_ok{device="None",cmd="QPGS2"} 1
LV6548_P1_is_ac_charging{device="None",cmd="QPGS2"} 0
LV6548_P1_is_scc_charging{device="None",cmd="QPGS2"} 1
LV6548_P1_is_battery_over_voltage{device="None",cmd="QPGS2"} 0
LV6548_P1_is_battery_under_voltage{device="None",cmd="QPGS2"} 0
LV6548_P1_is_line_lost{device="None",cmd="QPGS2"} 0
LV6548_P1_is_load_on{device="None",cmd="QPGS2"} 1
LV6548_P1_is_configuration_changed{device="None",cmd="QPGS2"} 0
LV6548_P1_output_mode{device="None",cmd="QPGS2",myStr="Phase 2 of 2 phase output (180°)"} 0
LV6548_P1_charger_source_priority{device="None",cmd="QPGS2",myStr="Utility first"} 0
LV6548_P1_max_charger_current{device="None",cmd="QPGS2"} 50
LV6548_P1_max_charger_range{device="None",cmd="QPGS2"} 120
LV6548_P1_max_ac_charger_current{device="None",cmd="QPGS2"} 20
LV6548_P1_pv1_input_current{device="None",cmd="QPGS2"} 1
LV6548_P1_battery_discharge_current{device="None",cmd="QPGS2"} 29
LV6548_P1_pv2_input_voltage{device="None",cmd="QPGS2"} 89.9
LV6548_P1_pv2_input_current{device="None",cmd="QPGS2"} 0`

Reply:
actually it is better to just remove the duplicate pop, ie remove these lines
```python
# build header
if "_command" in data:
    data.pop("_command")
```

Reply:
Done, tested, and working. Uploading now. Thank you!